// speechmatics/dist/types/models/notification-config.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notification_config_exports = {};
__export(notification_config_exports, {
  NotificationConfigContentsEnum: () => NotificationConfigContentsEnum,
  NotificationConfigMethodEnum: () => NotificationConfigMethodEnum
});
module.exports = __toCommonJS(notification_config_exports);
const NotificationConfigContentsEnum = {
  Jobinfo: "jobinfo",
  Transcript: "transcript",
  TranscriptJsonV2: "transcript.json-v2",
  TranscriptTxt: "transcript.txt",
  TranscriptSrt: "transcript.srt",
  Alignment: "alignment",
  AlignmentWordStartAndEnd: "alignment.word_start_and_end",
  AlignmentOnePerLine: "alignment.one_per_line",
  Data: "data",
  Text: "text"
};
const NotificationConfigMethodEnum = {
  Post: "post",
  Put: "put"
};
