// speechmatics/dist/types/models/model-error.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var model_error_exports = {};
__export(model_error_exports, {
  ModelErrorMessageEnum: () => ModelErrorMessageEnum,
  ModelErrorTypeEnum: () => ModelErrorTypeEnum
});
module.exports = __toCommonJS(model_error_exports);
const ModelErrorMessageEnum = {
  Error: "Error"
};
const ModelErrorTypeEnum = {
  InvalidMessage: "invalid_message",
  InvalidModel: "invalid_model",
  InvalidConfig: "invalid_config",
  InvalidAudioType: "invalid_audio_type",
  NotAuthorised: "not_authorised",
  InsufficientFunds: "insufficient_funds",
  NotAllowed: "not_allowed",
  JobError: "job_error",
  DataError: "data_error",
  BufferError: "buffer_error",
  ProtocolError: "protocol_error",
  UnknownError: "unknown_error"
};
