// speechmatics/dist/config/connection.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var connection_exports = {};
__export(connection_exports, {
  ConnectionConfigFull: () => ConnectionConfigFull
});
module.exports = __toCommonJS(connection_exports);
class ConnectionConfigFull {
  constructor(config) {
    this.batchUrl = "https://asr.api.speechmatics.com/v2";
    this.realtimeUrl = "wss://eu2.rt.speechmatics.com/v2";
    this.managementPlatformUrl = "https://mp.speechmatics.com/v1";
    if (typeof config === "string") {
      this.apiKey = config;
    } else {
      this.apiKey = config.apiKey;
      if (config.batchUrl)
        this.batchUrl = config.batchUrl;
      if (config.realtimeUrl)
        this.realtimeUrl = config.realtimeUrl;
      if (config.managementPlatformUrl)
        this.managementPlatformUrl = config.managementPlatformUrl;
      if (config.apiKey)
        this.apiKey = config.apiKey;
      this.appId = config.appId;
    }
  }
}
