// speechmatics/dist/types/models/audio-format-raw.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var audio_format_raw_exports = {};
__export(audio_format_raw_exports, {
  AudioFormatRawEncodingEnum: () => AudioFormatRawEncodingEnum,
  AudioFormatRawTypeEnum: () => AudioFormatRawTypeEnum
});
module.exports = __toCommonJS(audio_format_raw_exports);
const AudioFormatRawEncodingEnum = {
  PcmF32le: "pcm_f32le",
  PcmS16le: "pcm_s16le",
  Mulaw: "mulaw"
};
const AudioFormatRawTypeEnum = {
  Raw: "raw"
};
