// speechmatics/dist/types/models/batch-transcription-config.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var batch_transcription_config_exports = {};
__export(batch_transcription_config_exports, {
  BatchTranscriptionConfigDiarizationEnum: () => BatchTranscriptionConfigDiarizationEnum,
  BatchTranscriptionConfigMaxDelayModeEnum: () => BatchTranscriptionConfigMaxDelayModeEnum
});
module.exports = __toCommonJS(batch_transcription_config_exports);
const BatchTranscriptionConfigDiarizationEnum = {
  None: "none",
  Speaker: "speaker",
  Channel: "channel",
  SpeakerChange: "speaker_change",
  ChannelAndSpeakerChange: "channel_and_speaker_change"
};
const BatchTranscriptionConfigMaxDelayModeEnum = {
  Fixed: "fixed",
  Flexible: "flexible"
};
