// speechmatics/dist/types/models/realtime-message.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var realtime_message_exports = {};
__export(realtime_message_exports, {
  RealtimeMessageMessageEnum: () => RealtimeMessageMessageEnum
});
module.exports = __toCommonJS(realtime_message_exports);
const RealtimeMessageMessageEnum = {
  StartRecognition: "StartRecognition",
  AddAudio: "AddAudio",
  EndOfStream: "EndOfStream",
  SetRecognitionConfig: "SetRecognitionConfig",
  RecognitionStarted: "RecognitionStarted",
  AudioAdded: "AudioAdded",
  AddPartialTranscript: "AddPartialTranscript",
  AddTranscript: "AddTranscript",
  AddPartialTranslation: "AddPartialTranslation",
  AddTranslation: "AddTranslation",
  EndOfTranscript: "EndOfTranscript",
  AudioEventStarted: "AudioEventStarted",
  AudioEventEnded: "AudioEventEnded",
  Info: "Info",
  Warning: "Warning",
  Error: "Error"
};
